### PR TITLE
Status metrics on node start

### PR DESCRIPTION
**Pull request by iulianpascalau**
## Description of the reasoning behind the pull request 
- we needed resource statistics right from the start of the node. Also, network usage was missing from the periodic resource log prints.
  
## Proposed Changes
- refactored resourceMonitor and netStats implementation
- added a new component handler called statusCoreComponents that is initialized first on a node

## Testing procedure
- standard system test, grepping the log files on `node statistics` string. This string should be present on the first few lines of the log and repeated on exactly 30 seconds. We should monitor also the logs after shuffle out, we should not be able to see `node statistics` at intervals less than 30 seconds (that will show a programming error)

### CODE REVIEW

File: factory/statusCore/statusCoreComponentsHandler.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+		return err
+	}
+	mscc.statusCoreComponents = nil
+
+	return nil
+}
+
+// CheckSubcomponents verifies all subcomponents
+func (mscc *managedStatusCoreComponents) CheckSubcomponents() error {
+	mscc.mutCoreComponents.Lock()
```
Comment (sstanculeanu) [inline factory/statusCore/statusCoreComponentsHandler.go:70]:
```suggestion
	mscc.mutCoreComponents.RLock()
```

File: common/statistics/machine/netStatistics.go [hunk 2]
```diff
@@ -22,18 +20,47 @@ type netStatistics struct {
...
+	}
+
+	return stats
+}
+
+func (ns *netStatistics) startProcessLoop(handler func(ctx context.Context)) {
+	var ctx context.Context
+	ctx, ns.cancel = context.WithCancel(context.Background())
+
+	go handler(ctx)
```
Comment (sstanculeanu) [inline common/statistics/machine/netStatistics.go:46]: perhaps create 2 new constructors for tests only that have the handler as parameter? this way `newNetStatistics` won't be needed and this line would become `go ns.stats.processLoop(ctx)`

File: factory/statusCore/statusCoreComponentsHandler.go (hunk 1 above)
Comment (iulianpascalau) [reply factory/statusCore/statusCoreComponentsHandler.go:70]: done on all other components :+1:

File: common/statistics/machine/netStatistics.go (hunk 2 above)
Comment (iulianpascalau) [reply common/statistics/machine/netStatistics.go:46]: refactored

File: common/statistics/resourceMonitor.go [hunk 3]
```diff
@@ -163,32 +158,28 @@ func GetRuntimeStatistics() []interface{} {
...
-	for _, f := range files {
-		size += f.Size()
-	}
-
-	return core.ConvertBytes(uint64(size))
-}
-
 // SaveStatistics generates and saves statistic data on the disk
-func (rm *ResourceMonitor) SaveStatistics() {
+func (rm *resourceMonitor) SaveStatistics() {
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply common/statistics/resourceMonitor.go:162]: changed to `LogStatistics`

File: factory/statusCore/statusCoreComponentsHandler_test.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+import (
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go/factory/statusCore"
+	componentsMock "github.com/ElrondNetwork/elrond-go/testscommon/components"
+	"github.com/stretchr/testify/require"
+)
+
+// ------------ Test ManagedStatusCoreComponents --------------------
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply factory/statusCore/statusCoreComponentsHandler_test.go:11]: removed

[the remark this answers is not in the document]
Comment (iulianpascalau) [reply factory/statusCore/statusCoreComponentsHandler_test.go:11]: removed

**Review by miiu96 — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Pull request merged**